### PR TITLE
Add tests for canned-responses and redirects in VS/VSRs

### PR DESCRIPTION
**Pull request by tellet**
Cover testing scenarios for VS and VSRs.

### CODE REVIEW

File: tests/suite/test_v_s_route_redirects.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    def test_update(self, kube_apis, crd_ingress_controller, v_s_route_setup):
+        req_host = f"{v_s_route_setup.public_endpoint.public_ip}:{v_s_route_setup.public_endpoint.port}"
+        req_url_1 = f"http://{req_host}{v_s_route_setup.route_m.paths[0]}"
+        req_url_2 = f"http://{req_host}{v_s_route_setup.route_m.paths[1]}"
+        wait_before_test(1)
+        vs_name = f"{v_s_route_setup.namespace}/{v_s_route_setup.vs_name}"
+        vsr_name = f"{v_s_route_setup.namespace}/{v_s_route_setup.route_m.name}"
+        vsr_event_text = f"Configuration for {vsr_name} was added or updated"
+        vs_event_text = f"Configuration for {vs_name} was added or updated"
```
Comment (Dean-Coakley) [inline tests/suite/test_v_s_route_redirects.py]: Tiny optional suggestion:
I think it makes sense to swap the order of these lines.

File: tests/data/virtual-server-canned-responses/standard/virtual-server.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                 \"args\": \"${args}\",
+                 \"time_iso8601\": \"${time_iso8601}\",
+                 \"connections_reading\": \"${connections_reading}\"
+               }'
+  - path: "/canned-defaults"
+    action:
+      return:
+        code:
+        type:
+        body: >
```
Comment (pleshakov) [inline tests/data/virtual-server-canned-responses/standard/virtual-server.yaml]: this transforms into a string with one line "line1 line2 line3". if you want to test for multi-line string, I suggest changing to:
```yaml
body: |
  line 1
  line 2
  line 3
```

File: tests/suite/test_v_s_route_redirects.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        wait_before_test(1)
+        ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
+        config = get_vs_nginx_template_conf(kube_apis.v1,
+                                            v_s_route_setup.namespace,
+                                            v_s_route_setup.vs_name,
+                                            ic_pod_name,
+                                            ingress_controller_prerequisites.namespace)
+        assert 'return 307 ' in config and 'return 301 ' in config
+
+    def test_custom_canned_response(self, kube_apis, crd_ingress_controller, v_s_route_setup):
```
Comment (pleshakov) [inline tests/suite/test_v_s_route_redirects.py]: hm, the name is for canned responses but we're testing redirects here

File: tests/suite/test_v_s_route_redirects.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        req_host = f"{v_s_route_setup.public_endpoint.public_ip}:{v_s_route_setup.public_endpoint.port}"
+        req_url = f"http://{req_host}{v_s_route_setup.route_m.paths[0]}?arg1=arg"
+        wait_and_assert_status_code(307, req_url, v_s_route_setup.vs_host, allow_redirects=False)
+        resp = requests.get(req_url, headers={"host": v_s_route_setup.vs_host}, allow_redirects=False)
+        assert resp.headers['location'] == "http://example.com"
+
+    def test_default_canned_response(self, kube_apis, crd_ingress_controller, v_s_route_setup):
```
Comment (pleshakov) [inline tests/suite/test_v_s_route_redirects.py]: the name is for canned response, but we're testing for redirects here

File: tests/suite/test_v_s_route_redirects.py (hunk 3 above)
Comment (tellet) [reply tests/suite/test_v_s_route_redirects.py]: Oops, good point, thanks.

**Pull request merged**